src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -136,10 +138,65 @@ private void register(VillagerProfession prof, int id)
         professions.register(id, prof.name, prof);
     }
 
+    /**
+     * Register a trade to a {@link VillagerCareer} and a certain careerLevel (see {@link EntityVillager})
+     * 
+     * @param career
+     *            the villager career the trade is added to.
+     * @param careerLevel
+     *            the level a villager must have to offer this trade.
+     * @param trade
+     *            the trade that will be registered.
+     */
+    public void registerTrade(VillagerCareer career, int careerLevel, ITradeList trade)
+    {
+        Validate.notNull(trade, "A null trade cannot be registered");
+        List<List<ITradeList>> skilledTradeList = TRADES.get(career);
+        Validate.notNull(skilledTradeList, "The career '%s' of the profession '%s' has not yet been registered", career.name, career.profession.name);
+        if (careerLevel >= skilledTradeList.size())
+        {
+            skilledTradeList.add(careerLevel, null);
+        }
+        List<ITradeList> trades = skilledTradeList.get(careerLevel);
+        if (trades == null)
+        {
+            trades = new LinkedList<ITradeList>();
+            skilledTradeList.add(careerLevel, trades);
+        }
+        trades.add(trade);
+    }
+
+    /**
+     * Helper method to retrieve a {@link VillagerProfession} by name. The domain-part of the {@link ResourceLocation} refers to the mod adding the profession.
+     * 
+     * @param name
+     *            the ResourceLocation representing the profession
+     * @return the {@link VillagerProfession}, or, if not available, null
+     */
+    public VillagerProfession getVillagerProfession(ResourceLocation name)
+    {
+        return this.professions.getValue(name);
+    }
+
+    /**
+     * See {@link VillagerRegistry#getVillagerProfession(ResourceLocation)}. This function just converts the name into a ResourceLocation
+     * 
+     * @param name
+     *            the ResourceLocation representing the profession
+     * @return the {@link VillagerProfession}, or, if not available, null
+     */
+    public VillagerProfession getVillagerProfession(String name)
+    {
```

Comment:
No, they can access the registry itself and pull it out no need for bouncer methods.